pkg/executables/kubectl.go

```diff
@@ -382,6 +406,57 @@ func (k *Kubectl) Wait(ctx context.Context, kubeconfig string, timeout string, f
 	return nil
 }
 
+// CheckJsonPath will check a given JSONPath for a required state similar to wait command for objects without conditions.
+// This will be depreciated in favor of WaitJsonPath after version 1.23.
+func (k *Kubectl) CheckJsonPath(ctx context.Context, kubeconfig string, timeout string, jsonpath, forCondition string, property string, namespace string, opts ...KubectlOpt) error {
+	// On each retry kubectl wait timeout values will have to be adjusted to only wait for the remaining timeout duration.
+	//  Here we establish an absolute timeout time for this based on the caller-specified timeout.
+	timeoutDuration, err := time.ParseDuration(timeout)
+	if err != nil {
+		return fmt.Errorf("unparsable timeout specified: %v", err)
+	}
+	if timeoutDuration < 0 {
+		return fmt.Errorf("negative timeout specified: %v", err)
+	}
+	timeoutTime := time.Now().Add(timeoutDuration)
+
+	retrier := retrier.New(timeoutDuration, retrier.WithRetryPolicy(kubectlWaitRetryPolicy))
+	err = retrier.Retry(
+		func() error {
+			return k.checkJsonPath(ctx, kubeconfig, timeoutTime, jsonpath, forCondition, property, namespace, opts...)
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("executing wait: %v", err)
+	}
+	return nil
+}
+
+// CheckJsonPath will wait for a given JSONPath of a required state. Only compatible on K8s 1.23+
+func (k *Kubectl) WaitJsonPath(ctx context.Context, kubeconfig string, timeout string, jsonpath, forCondition string, property string, namespace string, opts ...KubectlOpt) error {
```

Comment:
We postponed adding this functionality due to the migration of kubectl executable to the kubernetes go client. We already have a temporary mechanism to fix this issue. @ewollesen thoughts?

Reply:
OK, found it, pkg/curatedpackages/packagecontrollerclient.go:107

So yeah, when installing the curated packages controller during cluster creation, it won't return until there's an active package.